packages/core/src/lib/Uuid.ts

```diff
@@ -2,6 +2,17 @@ import { randomUUID } from "crypto"
 
 export type Uuid<T> = string & { __brand: T }
 
+// Custom regex: admits any hex values matching the normal dashed cluster grouping structure
+// loose in that it does not attempt to reject 'impossible/undefined' based on the UUID spec
+const regex = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i
+
+function parse<T>(str: string): Uuid<T> {
+  if (!regex.test(str)) {
+    throw new Error(`Uuid: invalid format '${str}'`)
+  }
+  return str.toLowerCase() as Uuid<T>
```

Comment:
if you're going to canonicalize as lower case, better to do that immediately and validate that the thing you'll ultimately store is valid/compliant
(and, to be pedantic/emphasize that, I'd make the regeex not be case-insensitive)

Reply:
I don't really have a preference, so happy to go with that.